sdk/core/azure-core/src/http/response.cpp

```diff
@@ -9,13 +9,14 @@
 
 using namespace Azure::Core::Http;
 
-HttpStatusCode Response::GetStatusCode() { return m_statusCode; }
+HttpStatusCode Response::GetStatusCode() const { return m_statusCode; }
 
-std::string const& Response::GetReasonPhrase() { return m_reasonPhrase; }
+std::string const& Response::GetReasonPhrase() const { return m_reasonPhrase; }
 
-std::map<std::string, std::string> const& Response::GetHeaders() { return this->m_headers; }
+std::map<std::string, std::string> const& Response::GetHeaders() const { return this->m_headers; }
 
-std::vector<uint8_t>& Response::GetBodyBuffer() { return m_bodyBuffer; }
+const std::vector<uint8_t>& Response::GetBodyBuffer() const { return m_bodyBuffer; }
+std::vector<uint8_t>&& Response::MoveBodyBuffer() { return std::move(m_bodyBuffer); }
```

Comment:
Can you elaborate on the scenario that needs move semantics.  After calling this method the response will need to be discarded because all future callers will no longer have a valid response.

Reply:
Previously, a valid scenario is that the customer download some data, modify part of it and re-upload to the server, e.g. some page blob range modification scenario, by moving the buffer customer does not need to copy the buffer for upload.
However, this is no longer valid given the approach abandons buffer.

Reply:
so i guess this pr can be closed now?

Reply:
Not quite, there are still some immutable memeber function without `const` keyword. Please see latest commit.